### DISCUSSION TITLE
Cannot use the property which set  globally on the store with Setup Stores (Unable to use 'this' in arrow function)

### DISCUSSION THREAD

**Discussion by mangochu** [vuejs/pinia#2587] [Help and Questions] [answered]
Hello everyone!
I use the [setup store](https://pinia.vuejs.org/core-concepts/#Setup-Stores) to create my Pinia store.
But recently, I encountered some issues when I wanted to create properties for all stores as described in the documentation([Augmenting a Store](https://pinia.vuejs.org/core-concepts/plugins.html#Augmenting-a-Store)). I don't know how to call these properties in other store.

### **stores/index.ts**
```ts
import { createPinia } from 'pinia';
import { type Ref, markRaw, ref } from 'vue';
import { router } from '~/main';

const pinia = createPinia();
const globalLoading = ref(false);

pinia.use(({ store }) => {
	store.router = markRaw(router);
	store.globalLoading = globalLoading;
});

declare module 'pinia' {
	interface PiniaCustomProperties {
		router: typeof router
		globalLoading: Ref<boolean>;
	}
}

export default pinia;
```

### **main.ts**
```ts
import pinia from '~/stores/index';
...

createApp(App).use(pinia).use(router).mount('#app');
```

### **stores/auth.ts**
```ts
export const useAuthStore = defineStore('auth', () => {
...
// <- How can I call the `router` and `globalLoading` here?
})
```

How can I call the `router` and `globalLoading`  in the arrow function?
I found some answer that use `this.router`. However, 'this' does not work inside arrow functions.
Is there a similar method to import within defineStore, like...
```ts
defineStore('auth', (router, globalLoading) => { // <- Add parameters here
...
})
```

**Comment by posva** [comment, marked as answer]
- router: call `useRouter()` within the store
- `globalLoading`: just import it. This only works in SPA. If you need SSR to work too, use https://vuejs.org/api/composition-api-dependency-injection.html#composition-api-dependency-injection

**Comment by mangochu** [comment]
> * router: call `useRouter()` within the store
> * `globalLoading`: just import it. This only works in SPA. If you need SSR to work too, use https://vuejs.org/api/composition-api-dependency-injection.html#composition-api-dependency-injection

Thank you so much for your answer! I'm a Pinia newbie and I'm working hard to learn it. I wonder if there will be a chance to make some method like hook in the future for composition style. Thank you again.
